### PR TITLE
HTBHF-1377 Added webUIVersion to claim

### PR DESCRIPTION
**Pull request by YetAnotherMatt**

### CODE REVIEW

**Comment by YetAnotherMatt** [comment]
I'll have to re-number the sql script once the email address PR has been merged.

File: api/src/test/java/uk/gov/dhsc/htbhf/claimant/service/ClaimServiceTest.java [hunk 1]
```diff
@@ -347,9 +360,14 @@ void shouldHandleNullDeviceFingerprint() {
         var entitlement = new PaymentCycleVoucherEntitlement(asList(firstVoucherEntitlement, secondVoucherEntitlement));
         EligibilityAndEntitlementDecision decision = aDecisionWithStatusAndEntitlement(ELIGIBLE, entitlement);
         given(eligibilityAndEntitlementService.evaluateClaimant(any())).willReturn(decision);
+        ClaimRequest claimRequest = ClaimRequest.builder()
```
Comment (mikeSmith08) [inline api/src/test/java/uk/gov/dhsc/htbhf/claimant/service/ClaimServiceTest.java]: Looks like you could use a Test Data Factory here and for the two tests below too.

File: api/src/test/java/uk/gov/dhsc/htbhf/claimant/service/ClaimServiceTest.java [hunk 2]
```diff
@@ -462,6 +513,14 @@ void shouldSaveClaimantWhenEligibilityThrowsException() {
         assertThat(eventCaptor.getValue().getEventMetadata().get(FailureEvent.FAILED_EVENT_KEY)).isEqualTo(ClaimEventType.NEW_CLAIM);
     }
 
+    private ClaimRequest aClaimRequestForClaimant(Claimant claimant) {
```
Comment (mikeSmith08) [inline api/src/test/java/uk/gov/dhsc/htbhf/claimant/service/ClaimServiceTest.java]: Ah, this should be in a Test Data Factory so you can reuse above as per my other comment.

File: api/src/main/java/uk/gov/dhsc/htbhf/claimant/model/ClaimDTO.java [hunk 3]
```diff
@@ -28,4 +28,8 @@
     @ApiModelProperty(notes = "The fingerprint of the device used to make the claim, as best it can be identified. Probably a collection of header values.")
     private Map<String, Object> deviceFingerprint;
 
+    @JsonProperty("webUIVersion")
+    @ApiModelProperty(notes = "The version of the web-ui application used to create this request.")
```
Comment (mikeSmith08) [inline api/src/main/java/uk/gov/dhsc/htbhf/claimant/model/ClaimDTO.java]: Should this mention that it is used in conjunction with proving which version of the Ts and Cs they accepted?

File: api/src/test/java/uk/gov/dhsc/htbhf/claimant/service/ClaimServiceTest.java (hunk 1 above)
Comment (YetAnotherMatt) [reply api/src/test/java/uk/gov/dhsc/htbhf/claimant/service/ClaimServiceTest.java]: done

File: api/src/test/java/uk/gov/dhsc/htbhf/claimant/service/ClaimServiceTest.java (hunk 2 above)
Comment (YetAnotherMatt) [reply api/src/test/java/uk/gov/dhsc/htbhf/claimant/service/ClaimServiceTest.java]: done

File: api/src/main/java/uk/gov/dhsc/htbhf/claimant/model/ClaimDTO.java (hunk 3 above)
Comment (YetAnotherMatt) [reply api/src/main/java/uk/gov/dhsc/htbhf/claimant/model/ClaimDTO.java]: done

**Pull request merged**